test/jit/test_module_interface.py

```diff
@@ -202,6 +202,40 @@ def forward(self, x):
         with self.assertRaisesRegex(RuntimeError, "is not compatible with interface"):
             as_module_interface(scripted_wrong_mod)
 
+        # Check that interface implementations can be contravariant in argument types and covariant in return type.
+        global TensorToAny
+        @torch.jit.interface
+        class TensorToAny(nn.Module):
+            def forward(self, input: torch.Tensor) -> Any:
+                pass
+
+        @torch.jit.script
+        def as_tensor_to_any(x: TensorToAny) -> TensorToAny:
+            return x
+
+        global AnyToAny
+        @torch.jit.interface
+        class AnyToAny(nn.Module):
+            def forward(self, input: Any) -> Any:
+                pass
+
+        @torch.jit.script
+        def as_any_to_any(x: AnyToAny) -> AnyToAny:
+            return x
+
+        class TensorToAnyImpl(nn.Module):
+            def forward(self, input: Any) -> Any:
+                return input
+
+        class AnyToAnyImpl(nn.Module):
+            def forward(self, input: Any) -> torch.Tensor:
+                return torch.tensor([1])
+
+        as_tensor_to_any(torch.jit.script(TensorToAnyImpl()))
+        as_any_to_any(torch.jit.script(AnyToAnyImpl()))
+
```

Comment:
Optional:

It seems we don't have a test case where parameters are contravariant *and* results are covariant at the same time. I think it can be done via having a TensorToAnyImpl with f: Any->Tensor.